crates/storage/provider/src/traits/executor.rs

```diff
@@ -0,0 +1,45 @@
+//! Executor Factory
+
+use crate::{execution_result::ExecutionResult, StateProvider};
+use reth_interfaces::executor::Error;
+use reth_primitives::{Address, Block, ChainSpec, U256};
+
+/// Executor factory that would spawn the EVM with particular state provider.
```

Comment:
I would not use the terminology "spawn", build is more appropriate. Spawn sort of implies that it is a task spawned on a separate thread to me